code/go/0chain.net/smartcontract/minersc/miner.go

```diff
@@ -193,28 +193,20 @@ func (msc *MinerSmartContract) deleteNode(
 ) (*MinerNode, error) {
 	var err error
 	deleteNode.Delete = true
-	var nodeType spenum.Provider
 	switch deleteNode.NodeType {
 	case NodeTypeMiner:
-		nodeType = spenum.Miner
 	case NodeTypeSharder:
-		nodeType = spenum.Sharder
 	default:
 		return nil, fmt.Errorf("unrecognised node type: %v", deleteNode.NodeType.String())
 	}
 
-	for key, pool := range deleteNode.Pools {
+	for _, pool := range deleteNode.Pools {
 		switch pool.Status {
-		case spenum.Pending:
```

Comment:
miner/sharder ndoes do not have pending status anymore, they will be in active status when staked